File: backend/app/schemas/god.py
from __future__ import annotations

from datetime import datetime
from typing import Optional

from pydantic import BaseModel, Field


class GodBase(BaseModel):
    """神祇基础Schema"""
    name: str = Field(..., max_length=50, description="神祇名称")
    title: str = Field(..., max_length=100, description="神祇称号")
    role: str = Field(default="supreme", description="神祇角色")
    status: str = Field(default="active", description="神祇状态")
    description: Optional[str] = Field(None, description="神祇描述")
    power_level: int = Field(default=100, ge=1, le=100, description="神力等级")
    avatar: Optional[str] = Field(None, max_length=255, description="神祇头像URL")
    temple_name: Optional[str] = Field(None, max_length=100, description="神殿名称")
    blessing: Optional[str] = Field(None, description="神祇祝福语")
    is_eternal: bool = Field(default=True, description="是否永恒")
    reign_years: int = Field(default=0, ge=0, description="统治年数")


class GodCreate(GodBase):
    """创建神祇Schema"""
    pass


class GodUpdate(BaseModel):
    """更新神祇Schema"""
    title: Optional[str] = Field(None, max_length=100, description="神祇称号")
    status: Optional[str] = Field(None, description="神祇状态")
    description: Optional[str] = Field(None, description="神祇描述")
    power_level: Optional[int] = Field(None, ge=1, le=100, description="神力等级")
    avatar: Optional[str] = Field(None, max_length=255, description="神祇头像URL")
    temple_name: Optional[str] = Field(None, max_length=100, description="神殿名称")
    blessing: Optional[str] = Field(None, description="神祇祝福语")
    is_eternal: Optional[bool] = Field(None, description="是否永恒")
    reign_years: Optional[int] = Field(None, ge=0, description="统治年数")


class GodResponse(GodBase):
    """神祇响应Schema"""
    god_id: int
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None

    class Config:
        from_attributes = True


class GodListResponse(BaseModel):
    """神祇列表响应Schema"""
    total: int
    items: list[GodResponse]


class AdminLoginRequest(BaseModel):
    """管理员登录请求"""
    username: str = Field(..., max_length=50, description="管理员用户名")
    password: str = Field(..., min_length=6, description="管理员密码")


class AdminUserResponse(BaseModel):
    """管理员响应Schema"""
    admin_id: int
    username: str
    nickname: str
    role: str
    is_active: bool
    last_login: Optional[datetime] = None
    created_at: Optional[datetime] = None

    class Config:
        from_attributes = True


class AdminLoginResponse(BaseModel):
    """管理员登录响应"""
    access_token: str
    token_type: str = "bearer"
    admin: AdminUserResponse

File: frontend/services/god.ts
/**
 * 神祇管理 API 服务
 */
import axios from 'axios'

const API_BASE = import.meta.env.VITE_API_BASE_URL || '/api/v1'

export interface God {
  god_id: number
  name: string
  title: string
  role: string
  status: string
  description?: string
  power_level: number
  avatar?: string
  temple_name?: string
  blessing?: string
  is_eternal: boolean
  reign_years: number
  created_at?: string
  updated_at?: string
}

export interface GodListResponse {
  total: number
  items: God[]
}

export interface AdminUser {
  admin_id: number
  username: string
  nickname: string
  role: string
  is_active: boolean
  last_login?: string
  created_at?: string
}

export interface AdminLoginResponse {
  access_token: string
  token_type: string
  admin: AdminUser
}

/**
 * 获取神祇列表
 */
export async function getGods(skip = 0, limit = 100): Promise<GodListResponse> {
  const response = await axios.get<GodListResponse>(`${API_BASE}/god/`, {
    params: { skip, limit }
  })
  return response.data
}

/**
 * 获取单个神祇
 */
export async function getGod(godId: number): Promise<God> {
  const response = await axios.get<God>(`${API_BASE}/god/${godId}`)
  return response.data
}

/**
 * 管理员登录
 */
export async function adminLogin(username: string, password: string): Promise<AdminLoginResponse> {
  const formData = new URLSearchParams()
  formData.append('username', username)
  formData.append('password', password)
  
  const response = await axios.post<AdminLoginResponse>(
    `${API_BASE}/admin-auth/login`,
    formData,
    {
      headers: {
        'Content-Type': 'application/x-www-form-urlencoded'
      }
    }
  )
  return response.data
}

/**
 * 保存管理员令牌
 */
export function saveAdminToken(token: string) {
  localStorage.setItem('admin_token', token)
}

/**
 * 获取管理员令牌
 */
export function getAdminToken(): string | null {
  return localStorage.getItem('admin_token')
}

/**
 * 清除管理员令牌
 */
export function clearAdminToken() {
  localStorage.removeItem('admin_token')
}

/**
 * 检查是否已登录管理员
 */
export function isAdminLoggedIn(): boolean {
  return !!getAdminToken()
}
